### ISSUE TITLE
Feature request: Work as a pager when passing `.diff` or `.patch` files

### ISSUE THREAD

**Issue by marcelarie** [Wilfred/difftastic#793]
Maybe this is not the scope of the tool but I think it would be really nice to print the changes of a diff or patch file with `difftastic`.

Similar to how `colordiff` works:
<div align="center"><img width="500" src="https://github.com/user-attachments/assets/460ff941-b5c2-46ae-8dd2-4c80b06a883e" /></div>

**Comment by Wilfred** [comment]
Thanks for the suggestion! I'm afraid difftastic needs the original files so it can parse them, a .patch file isn't sufficient here.

**Comment by marcelarie** [comment]
> Thanks for the suggestion! I'm afraid difftastic needs the original files so it can parse them, a .patch file isn't sufficient here.

I am talking from the ignorance here, but couldn't it be possible if it treated each of the deletion and additions as files when reading from patch files? Comparing the deflation line to the addition and using the same logic that it uses when comparing files.

Thanks for the reply anyway!

**Comment by Wilfred** [comment]
True, but difftastic needs the whole file to parse e.g. all the matching braces. If you had a patch file with full context, it'd be possible, but usually patches just have a few surrounding lines.

This is exactly how difftastic handles conflict files though. It uses the conflict markers to construct the full original before and after.
